src/main/java/seedu/address/model/person/PersonSearchPredicate.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Name}, {@code Tags} or {@code Assets} matches any of the keywords given.
+ */
+public class PersonSearchPredicate implements Predicate<Person> {
+
+    private final List<String> keywords;
+
+    public PersonSearchPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                       .anyMatch(keyword -> {
+                           boolean nameContainsKeyword =
+                               StringUtil.containsWordIgnoreCase(person.getName().toString(), keyword);
+                           boolean tagsContainKeyword = person.getTags()
+                                .stream()
+                                .anyMatch(tag -> StringUtil.containsWordIgnoreCase(tag.get(), keyword));
+                           boolean assetsContainKeyword = person.getAssets()
+                                .stream()
+                                .anyMatch(asset -> StringUtil.containsWordIgnoreCase(asset.get(), keyword));
+                           return nameContainsKeyword || tagsContainKeyword || assetsContainKeyword;
```

Comment:
Is there a indentation misalignment here?

Reply:
It is intended as the lines would be too long otherwise.